### ISSUE TITLE
CellEditInit: Last Cell throws NumberFormatException if you push the Tabulator Button

### ISSUE THREAD

**Issue by enricokl** [primefaces/primefaces#8155]
**Describe the defect**
A dataTable in editMode "cell" and with the CellEditInit Ajax Event throws an NumberFormatException if you click tabulator in the last cell to be edited.

**Environment:**
 - PF Version: _8.0.13_
 - Affected browsers: _ALL_

**To Reproduce**
1. Click on last editable cell in the DataTable
2. Click on Tabulator
3. In log you see the NumberFormatException, because the value is "undefined"

**Expected behavior**
The EditMode ends without a Exception

**Example XHTML**
```html
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:p="http://primefaces.org/ui"
      xmlns:h="http://java.sun.com/jsf/html"
      xmlns:f="http://java.sun.com/jsf/core">

<h:head>
    <title>PrimeFaces Test</title>
</h:head>
<h:body>
    <h:form>
        <p:dataTable
                value="#{testView.getModels()}"
                editMode="cell"
                var="model"
                editable="true"

        >
            <p:ajax event="cellEditInit" listener="#{testView.onCellEditInit}"/>
            <p:ajax event="cellEdit" listener="#{testView.onCellEdit}"/>
            <p:ajax event="cellEditCancel" listener="#{testView.onCellEditCancel}"/>
            <p:column headerText="Test Col">
                <p:cellEditor>
                    <f:facet name="input">
                        <p:inputText value="#{model.value}"/>
                    </f:facet>
                    <f:facet name="output">
                        <h:outputText value="#{model.value}"/>
                    </f:facet>
                </p:cellEditor>
            </p:column>
        </p:dataTable>
    </h:form>
</h:body>
</html>
```

**Example Bean**
```java
package org.primefaces.test;

import org.primefaces.event.CellEditEvent;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Collection;

import javax.annotation.PostConstruct;
import javax.faces.view.ViewScoped;
import javax.inject.Named;


@Named
@ViewScoped
public class TestView implements Serializable {

   private ArrayList models = new ArrayList<>();

   @PostConstruct
   void init() {
      models.add(new TestModel("Eins"));
      models.add(new TestModel("Zwei"));
   }

   public Collection<TestModel> getModels() {
      return models;
   }

   public void onCellEditInit(CellEditEvent event) {

   }

   public void onCellEdit(CellEditEvent event) {

   }

   public void onCellEditCancel(CellEditEvent event) {

   }

}
```
```java
package org.primefaces.test;

import java.io.Serializable;


public class TestModel implements Serializable {

   private String value;

   TestModel() {
      this.value = "";
   }

   TestModel(final String value) {
      this.value = value;
   }

   public String getValue() {
      return this.value;
   }

   public void setValue(final String value) {
      this.value = value;
   }
}
```

The Problem is the Javascript in showCellEditor on line 4696 and getCellMeta on line 4903. There is no if-Condition, when the row is undefined.

There is a solution in version 10.x.x.

**Comment by tandraschko** [comment]
so its already fixed in 10.x?

**Comment by enricokl** [comment]
Yes, is it possible to fix this in 8.0 as well? If it's ok, I can fix it myself and do a PullRequest.

**Comment by tandraschko** [comment]
yea, thats the problem
this is the community issue tracker, so you cant provide a PR as elite releases are managed in  a private PrimeTek repo

i can just ping primetek here (cc @mertsincan)

otherwise PrimeFaces PRO user can push fixes directly in elite
